### PR TITLE
removed recursion from toHuman, solves #4

### PR DESCRIPTION
**Pull request by njelich**
this is a non breaking change, optimizing the function significantly and resolving #4

I ran yarn test and it passes all test cases

### CODE REVIEW

**Comment by njelich** [comment]
@chadoh Could you please take a look at this?

**Comment by willemneal** [comment]
He no longer has any permissions to approve this.

File: src/utils.ts [hunk 1]
```diff
@@ -167,29 +167,32 @@ export function toHuman(
   x: BN,
   baseUnit: string,
   magnitude: number,
-  adjustMagnitude = 0,
+  adjustMagnitude = 0
 ): string {
-  const nomination = new BN(10).pow(new BN(magnitude));
+  let adjustment;
```
Comment (willemneal) [inline src/utils.ts]: One nit here.  I would use a ternary here `let adjustment = <bool> ? 0 : Math...`.  And make `x.toString().length` a local variable to make it more readable.

**Review by willemneal — APPROVED** [review]
Just one nit. But otherwise great job!  No more endless recursion!

File: src/utils.ts (hunk 1 above)
Comment (njelich) [reply src/utils.ts]: sure, I'll edit it and push

**Comment by njelich** [comment]
@willemneal good now?

**Comment by willemneal** [comment]
Yep! But again I don't have any authority here.  @volovyk-s, @MaximusHaximus, or @ailisp might be able to help.

File: src/utils.ts [hunk 2]
```diff
@@ -167,29 +167,27 @@ export function toHuman(
   x: BN,
   baseUnit: string,
   magnitude: number,
-  adjustMagnitude = 0,
+  adjustMagnitude = 0
 ): string {
-  const nomination = new BN(10).pow(new BN(magnitude));
+  const numberOfDigits = x.toString().length;
+  const adjustment = numberOfDigits > magnitude ? 0 : Math.ceil((magnitude - x.toString().length + 1) / 3) * 3;
```
Comment (willemneal) [inline src/utils.ts]:
```suggestion
  const adjustment = numberOfDigits > magnitude ? 0 : Math.ceil((magnitude - numberOfDigits + 1) / 3) * 3;
```

**Review by volovyks — APPROVED** [review]

**Pull request merged**